utils/rjson.hh

```diff
@@ -145,6 +151,11 @@ inline rjson::value empty_string() {
 // The representation is dense - without any redundant indentation.
 std::string print(const rjson::value& value, size_t max_nested_level = default_max_nested_level);
 
+// Writes the JSON value to the output_stream, similar to print() -> string, but
+// directly. Note this has potentially more data copy overhead and should be 
+// reserved for larger values where not creating huge linear strings is useful.
```

Comment:
Both. But in the end, since we use small, temporary buffers, it is actually mainly copying that is the additional overhead. Though even this might be negligible, since serializing to string will probably cause a number of reallocs, which may or may not involve similar copies.  Do you want me to elaborate in the comment?

Reply:
Maybe a bit (mention that it may allocate additional memory of the size of the json).
This is an O() issue - it is obvious that print() must copy O(n) data and the question of whether it copies once or twice is less interesting in the O() sense. But the fact that this function **also** allocates O(n) memory is not obvious. It didn't have to be this way.

By the way, the fact it is this way is not all bad, and maybe if we explain/guarantee this to users  they can actually benefit from this: If I understand correctly, although print() returns a future, it actually completely reads the input json *before* returning the future. This means that the caller, after print() returns a future, may throw away the JSON! If this is in fact true, and if we guarantee that in a comment, the caller can use this knowledge to guarantee that although print() temporarily needs more memory (double) right away you can delete the original JSON and keep just the future.

Reply:
Ok, lengthened comment...